src/acvp.c

```diff
@@ -2236,6 +2259,84 @@ static ACVP_RESULT acvp_validate_test_session(ACVP_CTX *ctx) {
     return rv;
 }
 
+
+static
+ACVP_RESULT acvp_post_data(ACVP_CTX *ctx, char *filename) {
+    ACVP_RESULT rv = ACVP_SUCCESS;
+    JSON_Value *reg_arry_val = NULL;
+    JSON_Object *reg_obj = NULL;
+    JSON_Array *reg_arry = NULL;
+    JSON_Array *data_array = NULL;
+    JSON_Object *obj = NULL;
+    JSON_Value *val = NULL;
+    JSON_Value *post_val = NULL;
+    JSON_Value *raw_val = NULL;
+    const char *path = NULL;
+    char *json_result = NULL;
+
+    if (!ctx) {
+        return ACVP_NO_CTX;
+    }
+    if (!filename) {
+        ACVP_LOG_ERR("Must provide value for JSON filename");
+        return ACVP_MISSING_ARG;
+    }
+
+    if (strnlen_s(filename, ACVP_JSON_FILENAME_MAX + 1) > ACVP_JSON_FILENAME_MAX) {
+        ACVP_LOG_ERR("Provided filename length > max(%d)", ACVP_JSON_FILENAME_MAX);
+        return ACVP_INVALID_ARG;
+    }
+
+    val = json_parse_file(filename);
+    if (!val) {
+        ACVP_LOG_ERR("JSON val parse error");
+        return ACVP_MALFORMED_JSON;
+    }
+
+    data_array = json_value_get_array(val);
+    obj = json_array_get_object(data_array, 0);
+    if (!obj) {
+        ACVP_LOG_ERR("JSON obj parse error");
+        goto end;
+    }
+    path = json_object_get_string(obj, "url");
+    if (!path) {
+        ACVP_LOG_WARN("Missing path, POST aborted");
+        goto end;
+    }
+
+    raw_val = json_array_get_value(data_array, 1);
+    json_result = json_serialize_to_string_pretty(raw_val, NULL);
+    post_val = json_parse_string(json_result);
+    json_free_serialized_string(json_result);
+
+    rv = acvp_create_array(&reg_obj, &reg_arry_val, &reg_arry);
+    json_array_append_value(reg_arry, post_val);
+
+    json_result = json_serialize_to_string_pretty(reg_arry_val, NULL);
```

Comment:
You should use json_serialize_to_string because I modified the function to give the string length back as a parameter

Reply:
done